qa/L0_perf_resnet/run_test.sh

```diff
@@ -73,20 +66,58 @@ fi
 
 set +e
 
-# Run the model once to warm up. Some frameworks do optimization on the first requests.
-# Must warmup similar to actual run so that all instances are ready
-$PERF_CLIENT -v -i ${PERF_CLIENT_PROTOCOL} -m $MODEL_NAME -p${MEASUREMENT_WINDOW} \
-                -b${STATIC_BATCH} --concurrency-range ${CONCURRENCY}
+# Overload use of PERF_CLIENT_PROTOCOL for convenience with existing test and 
+# reporting structure, though "triton_c_api" is not strictly a "protocol".
+# NOTE: tritonserver will be ran in-process with perf_analyzer
+if [[ "${PERF_CLIENT_PROTOCOL}" == "triton_c_api" ]]; then
+    # Using C API requires extra info to start in-process server
+    # NOTE: PA C API doesn't expose --backend-directory, but defaults to
+    #       ${TRITON_DIR}/backends
+    # NOTE: PA C API doesn't expose --backend-config, but this is only used
+    #       for tfserving grpc test currently.
+    SERVICE_ARGS="--service-kind triton_c_api \
+                  --triton-server-directory ${TRITON_DIR} \
+                  --model-repository ${MODEL_REPO}"
+# Otherwise run server as separate process as normal via HTTP/GRPC
```

Comment:
No need to add so many comments. The if-else logic is simple enough to follow. A lot of comments sometimes becomes difficult to maintain. Imagine in future if a new condition is added and we skip updating this comment which is already in if block.
If you think logic is being hidden then you can do something like:
```
if [[ "${PERF_CLIENT_PROTOCOL}" == "triton_c_api" ]]; then
....
else # PERF_CLIENT_PROTOCOL != "triton_c_api"
...
fi
```